packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -411,6 +413,36 @@ class ChromiumLauncher {
     return chrome;
   }
 
+  /// Gets the first [chrome] tab.
+  ///
+  /// Note: Retry getting tabs from Chrome for a few seconds and retry finding
+  /// the tab a few times. This prevents flakes caused by Chrome not returning
+  /// correct output if the call was too close to the start.
+  Future<ChromeTab?> _getFirstTab(Chromium chrome) async {
+    const Duration retryFor = Duration(seconds: 2);
```

Comment:
This is intended to retry for up to 10s (2s * 5 tries)?

FYI, the `getTabs(retryFor)` call automatically retries around 4 times, and for as long as the given duration. This method - which may be intended? - is then layered more retry efforts on top of that.

https://github.com/google/webkit_inspection_protocol.dart/pull/88/files#diff-1c32e5460556de493e3a74667b36726abd37a69815a1e579cd18a5a1eac9b54fR47

Reply:
The layers of retries are intentional - normally we would retry until we get the first tab (the first attempts might return empty tab list).

In theory, we should always succeed, and previously the tool crashed at the first failure so it is hard to tell how many times the current failure can happen in a row. 

I might have added too many retries, but I would like to collect more information if the failures happen again - if we cannot connect to chrome for that long then something is else is going wrong and needs investigation. I would also change it to less retries at that moment to make sure we do not delay the tests too much.

Reply:
Gotcha! Sounds like this is indeed what's intended.